getters.go

```diff
@@ -14,66 +14,84 @@ import (
 // getterCache caches the results for all the getters in this file.
 var getterCache = cache.New(time.Second*90, time.Minute*5)
 
-func FindCachedAgent(ctx DBContext, agentID string) (*models.Agent, error) {
-	if agentID == uuid.Nil.String() {
+func FindCachedAgent(ctx DBContext, id string) (*models.Agent, error) {
+	if id == uuid.Nil.String() {
 		return nil, nil
 	}
 
-	agent, err := findCachedEntity[models.Agent](ctx, agentID)
+	agent, err := findCachedEntity[models.Agent](ctx, id)
 	if err != nil {
 		return nil, err
 	}
 
 	return agent, nil
 }
 
-func FindCachedCheck(ctx DBContext, agentID string) (*models.Check, error) {
-	check, err := findCachedEntity[models.Check](ctx, agentID)
+func FindCachedCheck(ctx DBContext, id string) (*models.Check, error) {
+	check, err := findCachedEntity[models.Check](ctx, id)
 	if err != nil {
 		return nil, err
 	}
 
 	return check, nil
 }
 
-func FindCachedCanary(ctx DBContext, agentID string) (*models.Canary, error) {
-	canary, err := findCachedEntity[models.Canary](ctx, agentID)
+func FindCachedCanary(ctx DBContext, id string) (*models.Canary, error) {
+	canary, err := findCachedEntity[models.Canary](ctx, id)
 	if err != nil {
 		return nil, err
 	}
 
 	return canary, nil
 }
 
-func FindCachedPerson(ctx DBContext, agentID string) (*models.Person, error) {
-	person, err := findCachedEntity[models.Person](ctx, agentID)
+func FindCachedPerson(ctx DBContext, id string) (*models.Person, error) {
```

Comment:
Why can't we just have FindPerson and FindTeam and then it checks whether it is a uuid, string or email ?

Reply:
Do we want to have the option to optionally not use cache? I think cached getter is mainly for notification templating.